hibernate-core/src/main/java/org/hibernate/dialect/DB2Dialect.java

```diff
@@ -356,6 +357,11 @@ public void initializeFunctionRegistry(QueryEngine queryEngine) {
 				new DB2FormatEmulation( queryEngine.getTypeConfiguration() )
 		);
 
+		queryEngine.getSqmFunctionRegistry().register(
+				"atan2",
+				new DB2Atan2Function( queryEngine.getTypeConfiguration() )
+		);
```

Comment:
While testing I noticed that DB2 requires the parameters to the `atan2` function in reverse order. This change fixes it, don't know if there's a better/more standard way of doing this.

Reply:
Usually we use a pattern for this purpose:

```
queryEngine.getSqmFunctionRegistry().patternDescriptorBuilder("atan2", "atan2(?2,?1)")
	.setInvariantType( queryEngine.getTypeConfiguration().getBasicTypeRegistry().resolve( StandardBasicTypes.DOUBLE ) )
	.setExactArgumentCount( 2 )
	.setParameterTypes(NUMERIC, NUMERIC )
	.register();
```

Reply:
Thanks for the tip, I thought there was a better way to do it. Updated it now.